experimental/dds/tree2/src/test/feature-libraries/objectForest.spec.ts

```diff
@@ -80,34 +80,18 @@ describe("object-forest", () => {
 		});
 	});
 
-	describe("moveCursorToPath", () => {
-		it("moves cursor to specified path.", () => {
-			const forest = buildForest();
-			initializeForest(forest, [singleJsonCursor([1, 2])]);
-
-			const cursor = forest.allocateCursor();
-			const path: UpPath = {
-				parent: undefined,
-				parentField: rootFieldKey,
-				parentIndex: 0,
-			};
-
-			forest.moveCursorToPath(path, cursor);
-			assert.deepEqual(path, cursor.getPath());
-		});
-		it("undefined path points to dummy node above detachedFields.", () => {
-			const forest = buildForest();
-			initializeForest(forest, [singleJsonCursor([1, 2])]);
-			const cursor = forest.allocateCursor();
-			forest.moveCursorToPath(undefined, cursor);
-			assert.deepEqual(cursor.getFieldKey(), singleMapTreeCursor(forest.roots).getFieldKey());
-		});
+	it("moveCursorToPath with an undefined path points to dummy node above detachedFields.", () => {
+		const forest = buildForest();
+		initializeForest(forest, [singleJsonCursor([1, 2])]);
+		const cursor = forest.allocateCursor();
+		forest.moveCursorToPath(undefined, cursor);
+		assert.deepEqual(cursor.getFieldKey(), singleMapTreeCursor(forest.roots).getFieldKey());
 	});
 
 	it("getCursorAboveDetachedFields", () => {
 		const forest = buildForest();
 		initializeForest(forest, [singleJsonCursor([1, 2])]);
 		const cursor = forest.getCursorAboveDetachedFields();
-		assert.deepEqual(cursor.getFieldKey(), singleMapTreeCursor(forest.roots).getFieldKey());
+		assert.deepEqual(cursor, singleMapTreeCursor(forest.roots));
```

Comment:
deepEqual doesn't really do the correct whing when comparing cursors. This might happen to work right now, but could easily break if for example the implementation changed to a different cursor implementation.

Its better to check if the cursors expose the correct tree content.

For example save the jsonable tree you used to a local, then copy the cursor into a jsonable tree, then compare those two jsonable trees with deep equal rather than comparing the cursor objects.

Reply:
Updated test to compare jsonableTrees instead of cursors